src/types/resolved.ts

```diff
@@ -1,6 +1,7 @@
 import { Param } from '@/types/params'
+import { Route } from '@/types/routes'
 
-export type RouteFlat = {
+export type Resolved<T extends Route> = T & {
```

Comment:
We could probably use `T` to tell us what `path` and `params` actually are. I don't think there's much of a benefit to `path` but maybe to `params`? No change requested. Just an idea.

Reply:
On a separate note. Doesn't really seem like we're using `T` for anything. Should this just be `Route & { ... }`?

Reply:
moved T into `matched` property, I still prefer having the explicitness of how this is written over `Resolved` just being a standalone type